### DISCUSSION TITLE
How do you add roles to firebase sign in with google or facebook

### DISCUSSION THREAD

**Discussion by brianpooe** [hasura/graphql-engine#7860] [General]
Is it possible to add roles to the jwt token for firebase sign in with google or facebook?

For instance i have my hasura roles to access specific queries and mutations but only signup with password works cause an endpoint is created on the hasura actions but sign up with social platforms happens on the frontend.

**Comment by johnclary** [comment]
@brianpooe I'm only familiar with Auth0, but the general concept would be the same. You would ideally configure a hook in your auth provider to upsert a user mutation when then user signs in. That approach is described [here](https://hasura.io/docs/latest/graphql/core/guides/integrations/auth0-jwt.html).
